src/PointwiseFunctions/GeneralRelativity/WeylPropagating.hpp

```diff
@@ -0,0 +1,71 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace gsl {
+template <typename>
+struct not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace gr {
+// @{
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes the propagating modes of Weyl tensor
+ *
+ * \details The Weyl tensor evolution system in vacuum has six characteristic
+ * fields, of which two (\f$ U^{8\pm}\f$) are proportional to the
+ * Newman-Penrose components of the Weyl tensor \f$\Psi_4\f$ and \f$\Psi_0\f$.
+ * These represent the true gravitational-wave degrees of freedom, and
+ * can be written down in terms of \f$3+1\f$ quantities as
+ * \cite Kidder2004rw (see Eq. 75).
+ * \f{align*}
+ * U^{8\pm}_{ij} &= \left(P^{k}_i P^{l}_j - \frac{1}{2} P_{ij} P^{kl}\right)
+ *          \left(R_{kl} + K K_{kl} - K_k^m K_{ml}
+ *                \mp n^m \nabla_m K_{kl} \pm n^m \nabla_{(k}K_{l)m}\right),\\
+ *  &= \left(P^{k}_i P^{l}_j - \frac{1}{2} P_{ij} P^{kl}\right)
+ * \left(E_{kl} \mp n^m \nabla_m K_{kl} \pm n^m \nabla_{(k}K_{l)m}\right),
+ * \f}
+ * where \f$R_{ij}\f$ is the spatial Ricci tensor, \f$K_{ij}\f$ is the
+ * extrinsic curvature, \f$K\f$ is the trace of \f$K_{ij}\f$,  \f$n^i\f$
+ * is a spatial unit vector normal to the interface, and \f$P^{ij}\f$ and its
+ * index-raised and lowered forms project tensors transverse to \f$n^i\f$.
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ii<DataType, SpatialDim, Frame> weyl_propagating(
+    const tnsr::ii<DataType, SpatialDim, Frame>& ricci,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::ijj<DataType, SpatialDim, Frame>& cov_deriv_extrinsic_curvature,
+    const tnsr::I<DataType, SpatialDim, Frame>& unit_interface_normal_vector,
+    const tnsr::II<DataType, SpatialDim, Frame>& projection_IJ,
+    const tnsr::ii<DataType, SpatialDim, Frame>& projection_ij,
+    const tnsr::Ij<DataType, SpatialDim, Frame>& projection_Ij,
+    const short sign) noexcept;
```

Comment:
I'm not sure `short` is better or worse than `double`. Not a change request, just sort of thinking "out loud". I have a slight preference toward `double`, though

Reply:
... `short` uses a quarter of the memory of `double` but does require explicit casts later when its multiplied with double. Am not sure which wins, but maybe the tiny memory save is not worth the multiple runtime casts.

Reply:
Yea, I have no idea... Since it's a temporary and only a single value, `double` seems fine :shrug: but I really have no idea :P Probably the only argument one way or the other that holds any merit is "which is more intuitive" :shrug: